backend/src/backend/db/session.py

```
@@ -0,0 +1,34 @@
+        "postgresql", "postgresql+psycopg"
+    )
+
+    # recycle connections after 5 minutes
+    # to correspond with the compute scale down
+    engine = create_engine(
+        connection_string,
+        connect_args={"sslmode": "require"},
+        pool_recycle=300
+    )
+except ValueError as e:
+    raise RuntimeError(f"Failed to initialize database connection: {str(e)}")
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)